### PR TITLE
Feat/upgrade mui v5

### PR DESCRIPTION
**Pull request by HaleyWardo**
I linked this to PHC and ran into almost no issues. The breaking changes were easy to resolve. [Here is a good first pass at those changes](https://github.com/lifeomic/phc-ui/compare/TESTING-chroma-mui-v5?expand=1). There is still more todo - like updating all existing MUI imports to use `@mui/material` and the fade/alpha replace.

**Changes**
- Upgrade MUI to v5 [[migration docs](https://mui.com/material-ui/migration/migration-v4/)]
- Added resolution for JSDOM since fireEvents `blur` and `focus` events weren't firing with new react/react-dom update [[issue](https://github.com/testing-library/user-event/issues/592)].
- **A LOT** of type updates on the theme overrides
- Added another create theme override snapshot

**Dependency updates**
 - ❌ Removed `material-ui/core`,  `material-ui/styles`, and `material-ui/types`
 - ➕ Added `mui/material`, `mui/styles`,  `emotion/react`, and `emotion/styled`
   - We will remove `mui/styles` once we move to the new styling solution
 - ⬆ Updated `react` and `react-dom` versions

**Breaking changes**
1. MUI did a re-branding and renamed the packages from `@material/*` to `@mui/*` so we will need to update to use the new peer dependencies below. @emotion-react is a MUI peer dependency for their styled components. 

```diff
-     "@material-ui/core": "^4.9.11",
-     "@material-ui/styles": "^4.9.10",
+     "@mui/material": "^5.9.3",
+     "@mui/styles": "^5.9.3",
+     "@emotion/react": "^11.9.3",
```
2. React versions needs updated to >= `16.14.0`. Their [docs](https://mui.com/material-ui/migration/v5-style-changes/#restructure-component-definitions) state that the minimum version is `17` but I believe that is if you are using their styled components. In another focused PR, I will work on updating us to react 17. 
3. The `fade` utility is now `alpha` [[docs](https://mui.com/material-ui/migration/v5-style-changes/#mui-material-styles)].

```diff
-    import fade from '@material-ui/core/fade`;
+    import { alpha } from '@mui/material/styles`;
```
4. The theme spacing now returns pixel values instead of a number so all instances where we add `px` after `theme.spacing`, will need to be removed [[docs](https://mui.com/material-ui/migration/v5-style-changes/#%E2%9C%85-remove-px-suffix)].
```diff
-    gap: `${theme.spacing(1)}px`;
+    gap: theme.spacing(1);
```
5. The create theme overrides has been restructured [[docs](https://mui.com/material-ui/migration/v5-style-changes/#restructure-component-definitions)].
```diff
const theme = createTheme({
-  overrides: {
-    MuiButton: {
-      root: { padding: 0 },
-    },
-  },
+  components: {
+    MuiButton: {
+      styleOverrides: {
+        root: { padding: 0 },
+      },
+    },
+  },
 });
```

### CODE REVIEW

**Comment by loscm** [comment]
Requested review from a random `@lifeomic/security` team member, @bishopb, due to term: **root**.

This review is not blocking and is for broader awareness.  Consider if this change requires deeper security review and ask when it is necessary.

- [Automation](https://github.com/lifeomic/pr-security-flag-scanner/blob/master/src/lambdas/pr-security-flag-scanner/index.ts) | [#pr-flagged-sec-review](https://lifeomic.slack.com/messages/pr-flagged-sec-review)

**Comment by xdumaine** [comment]
Wow excellent PR writeup. I had no idea about the `diff` codeblock type.

File: src/typeUtils.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 // eslint-disable-next-line no-restricted-imports
```
Comment (xdumaine) [inline src/typeUtils.ts:1]: I think this can be nixed.

**Review by xdumaine — APPROVED** [review]

**Review by swain — COMMENTED** [review]
Changes LGTM, other than the breaking change thing I mentioned.

**Review by swain — APPROVED** [review]
:shipit:

**Review by xdumaine — APPROVED** [review]

**Review by AClineff — APPROVED** [review]
Noice.

**Pull request merged**